astroid/brain/brain_dataclasses.py

```diff
@@ -503,6 +503,15 @@ def _looks_like_dataclass_field_call(
     return inferred.name == FIELD_NAME and inferred.root().name in DATACLASS_MODULES
 
 
+def _looks_like_dataclasses(node: nodes.Module) -> bool:
+    return node.qname() == "dataclasses"
+
+
+def _resolve_private_replace_to_public(node: nodes.Module) -> None:
+    if "_replace" in node.locals:
+        node.locals["replace"] = node.locals["_replace"]
```

Comment:
Could you explain this change? Doesn't that make us incompatible with people adding their own `replace` method?

Reply:
In python/cpython@6f3c138dfa868b32d3288898923bbfa388f2fa5d, a `_replace()` method was [extracted](https://github.com/python/cpython/blob/6f3c138dfa868b32d3288898923bbfa388f2fa5d/Lib/dataclasses.py#L1552) from `replace()`. Astroid became unable to infer the result of `replace()` due to this added indirection.

> Doesn't that make us incompatible with people adding their own replace method?

I'm not sure, but if so, isn't this also a problem with most/all brain transformations?

Reply:
Ah, make sense! Could you add a comment with a link to that commit here?